flink-test-utils-parent/flink-test-utils-junit/src/test/java/org/apache/flink/testutils/junit/RetryOnFailureTest.java

```diff
@@ -18,17 +18,16 @@
 
 package org.apache.flink.testutils.junit;
 
-import org.apache.flink.testutils.junit.extensions.retry.RetryExtension;
-
-import org.junit.jupiter.api.AfterAll;
-import org.junit.jupiter.api.TestTemplate;
-import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.AfterClass;
+import org.junit.Rule;
+import org.junit.Test;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
-/** Tests for the RetryOnFailure annotation. */
-@ExtendWith(RetryExtension.class)
-class RetryOnFailureTest {
+/** Tests for the {@link RetryOnFailure} annotation on JUnit4 {@link RetryRule}. */
+public class RetryOnFailureTest {
```

Comment:
@RyanSkraba sorry for reiterating over the PR again but could you also move this change out into its own commit? Initially, that's what I wanted to do myself. But I realized that I probably would overwrite your authorship.

Reply:
No problem -- you've been very reactive, thanks!

I assume you mean that you want all of the changes back to JUnit4 to be in a separate commit.  I'll prepare that for RetryOnExceptionTest as well.

Reply:
Sorry for responding that late and being picky about it. But I think we should improve the commit messages of the commits to make it clearer where they belong to. The JavaDoc commit is actually a documentation improvement. Therefore, the commit message should start like `[hotfix][docs] Clarify javadoc exceptions`. The revert of the JUnit5 migration is actually related to FLINK-27740. It was unintentionally introduced there. Hence, I'd recommend a commit message like 
```
[FLINK-27740][test] Revert JUnit5 migration for RetryOnFailureTest

It's actually meant to test the JUnit4 RetryRule feature. The corresponding JUnit5 Extension is tested in `RetryOnFailureExtensionTest`.
```
The FLINK-27740 commit needs to be backported to 1.16 as a consequence to ensure proper test coverage in the `release-1.16` branch.